From 42b5b17cfe36405a348104403caa8aa45db6dc78 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Mon, 23 May 2016 05:03:46 -0400
Subject: [PATCH 26/62] fix quoted empty string failed

While password or username is a empty string, it failed to
invoke quote:
...
|  File "/usr/lib64/python2.7/site-packages/pyanaconda/iutil.py",
line 823, in parse_components
|    quote(self.password) or "")
|  File "/usr/lib64/python2.7/urllib.py", line 1290, in quote
|    raise TypeError('None object cannot be quoted')
...

Upstream-Status: Pending

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 pyanaconda/iutil.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/pyanaconda/iutil.py b/pyanaconda/iutil.py
index 7c4e435..0e18bbf 100644
--- a/pyanaconda/iutil.py
+++ b/pyanaconda/iutil.py
@@ -819,9 +819,13 @@ class ProxyString(object):
     def parse_components(self):
         """ Parse the components of a proxy url into url and noauth_url
         """
-        if self.username or self.password:
+        if self.username and self.password:
             self.proxy_auth = "%s:%s@" % (quote(self.username) or "",
                                           quote(self.password) or "")
+        elif self.username and not self.password:
+            self.proxy_auth = "%s:@" % (quote(self.username) or "")
+        elif not self.username and self.password:
+            self.proxy_auth = ":%s@" % (quote(self.password) or "")
 
         self.url = self.protocol + self.proxy_auth + self.host + ":" + self.port
         self.noauth_url = self.protocol + self.host + ":" + self.port
-- 
2.7.4

